Core/clim-basic/extended-streams/text-formatting.lisp

```diff
@@ -31,23 +31,25 @@
 
 (defclass standard-page-layout ()
   ((%page-region :reader stream-page-region :writer (setf %page-region))
-   (margins :initform '(:left   (:absolute 0)
-                        :top    (:absolute 0)
+   (margins :initform '(:left   (:relative 0)
+                        :top    (:relative 0)
                         :right  (:relative 0)
                         :bottom (:relative 0))
             :accessor stream-text-margins :type margin-spec)))
 
 (defmethod initialize-instance :after ((instance standard-page-layout)
                                        &key text-margins text-margin)
   (macrolet ((thunk (edge default)
-               `(unless (getf text-margins ,edge)
-                  (setf (getf text-margins ,edge) ,default))))
-    (thunk :left   '(:absolute 0))
-    (thunk :top    '(:absolute 0))
+               `(let ((spec (getf text-margins ,edge)))
+                  (typecase spec
```

Comment:
Since this is repeated below, add a `normalize-margin-spec` or similar helper function.

Reply:
it is slightly different (below macrolet merges old and new specs while this one provides defaults)

Reply:
As far as I can tell, a function `normalize-margin-spec-plist plist defaults` called as

* `(normalize-margin-spec-plist text-margins '(:left (:absolute 0) :top (:absolute 0) …))`

* `(normalize-margin-spec-plist text-margins old-margins)`

would do the job.

Reply:
done

Reply:
done